aiida_lsmo/workchains/isotherm_accurate.py

```diff
@@ -512,33 +514,35 @@ def should_run_another_gcmc_highp(self):
         """
         if self.ctx.first_iteration:  # Proceed with first iteration
             self.ctx.first_iteration = False
-            self.ctx.psat = self.ctx.parameters['p_sat_coeff'] * self.ctx.parameters['loading_highp_sigma'] * self.ctx.geom['Estimated_saturation_loading'] / (1 
-                    - self.ctx.parameters['loading_highp_sigma']) / self.ctx.kh / 100000 # the pressure in in bar now
+            self.ctx.psat = self.ctx.parameters['p_sat_coeff'] * self.ctx.parameters[
+                'loading_highp_sigma'] * self.ctx.geom['Estimated_saturation_loading'] / (
+                    1 - self.ctx.parameters['loading_highp_sigma']) / self.ctx.kh / 100000  # the pressure in in bar now
             self.ctx.dpmax = (self.ctx.psat - self.ctx.ph0
                              ) / self.ctx.parameters['n_dpmax']  # Comment: maybe more efficient to use the first ph0
             self.ctx.pressure_old = self.ctx.pressure
-            self.ctx.pressure = self.ctx.pressure_old + 10 * (self.ctx.ph0 / self.ctx.psat) * (self.ctx.geom['Estimated_saturation_loading'] / self.ctx.kh / 100000)
+            self.ctx.pressure = self.ctx.pressure_old + 10 * (self.ctx.ph0 / self.ctx.psat) * (
+                self.ctx.geom['Estimated_saturation_loading'] / self.ctx.kh / 100000)
             self.ctx.gcmc_i += 1
             return True
-        else:
-            gcmc_loading_average_old = self.ctx.gcmc_loading_average
-            self.ctx.gcmc_loading_average = self._get_last_loading_molkg()
-            loading_derivative = (self.ctx.gcmc_loading_average - gcmc_loading_average_old) / (self.ctx.pressure - self.ctx.pressure_old)
-            if loading_derivative < 0:  #Need to recompute GCMC. NOTE: very dangerous, as it can lead to infinite loop!
-                self.report('WARNING: recomputing same pressure point, because loading has negative derivative.')
-                self.ctx.gcmc_loading_average = gcmc_loading_average_old
-                del self.ctx.raspa_gcmc[-1]
-                return True
-            dp_computed = 10 * self.ctx.pressure / self.ctx.psat * self.ctx.geom['Estimated_saturation_loading'] / loading_derivative
-            self.ctx.pressure_old = self.ctx.pressure
-            self.ctx.pressure = self.ctx.pressure_old + min(self.ctx.dpmax, dp_computed)
-            if self.ctx.pressure > self.ctx.psat:
-                return False # Converged
-            else:
-                self.ctx.gcmc_i += 1
-                return True # Didn't converge yet
 
-        return self.ctx.current_p_index < len(self.ctx.pressures)
```

Comment:
@danieleongari This line was not reachable and I've therefore removed it.
Let me know if that was unexpected